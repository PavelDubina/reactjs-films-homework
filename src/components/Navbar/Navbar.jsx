/*
 * Copyright Â© 2020 EPAM Systems, Inc. All Rights Reserved. All information contained herein is, and remains the
 * property of EPAM Systems, Inc. and/or its suppliers and is protected by international intellectual
 * property law. Dissemination of this information or reproduction of this material is strictly forbidden,
 * unless prior written permission is obtained from EPAM Systems, Inc
 */

import React, { useState } from 'react'
import { Select } from '../common/Select'
import className from 'classnames'
import styles from './Navbar.scss'
import response from '../../utils/genres.json'
import { navBtns } from './../../utils/constants'

export const Navbar = () => {
  const [active, setActive] = useState(0)
  return (
    <div className={styles.navbar}>
      <div className={styles.navbar__buttons}>
        {navBtns.map((button, i) => {
          const key = i
          const classStyle = className(styles.navbar__buttons_button, {
            [styles.active_button]: active === key,
          })
          return (
            <div className={classStyle} onClick={() => setActive(key)} key={key}>
              {button}
            </div>
          )
        })}
        <Select genres={response.genres} />
      </div>
    </div>
  )
}
