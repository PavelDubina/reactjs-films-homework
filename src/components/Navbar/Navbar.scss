/*
 * Copyright Â© 2020 EPAM Systems, Inc. All Rights Reserved. All information contained herein is, and remains the
 * property of EPAM Systems, Inc. and/or its suppliers and is protected by international intellectual
 * property law. Dissemination of this information or reproduction of this material is strictly forbidden,
 * unless prior written permission is obtained from EPAM Systems, Inc
 */
.navbar {
  &__buttons {
    &_button {
      color: rgb(24, 183, 236);
      background-color: rgb(240, 240, 240);
      margin: 0 20px 0 0;
      padding: 0;
      width: fit-content;
      border-radius: 0;
    }

    &_select {
      border: none;
      font-size: 16px;
      color: rgb(24, 183, 236);
      background-color: rgb(240, 240, 240);
      outline: none;

      option:checked {
        background-color: rgb(24, 183, 236);
        color: #fff;
      }
    }
  }
}

.active {
  border-bottom: 1px solid rgb(24, 183, 236);
}
