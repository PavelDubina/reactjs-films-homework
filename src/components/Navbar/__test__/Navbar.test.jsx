/*
 * Copyright Â© 2020 EPAM Systems, Inc. All Rights Reserved. All information contained herein is, and remains the
 * property of EPAM Systems, Inc. and/or its suppliers and is protected by international intellectual
 * property law. Dissemination of this information or reproduction of this material is strictly forbidden,
 * unless prior written permission is obtained from EPAM Systems, Inc
 */

import React from 'react'
import { create, act } from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { Navbar } from '../Navbar'

describe('Navbar', () => {
  const mockStore = configureStore([])
  let store
  let component
  beforeEach(() => {
    store = mockStore({
      genres: { genres: [{ id: 1, name: 'test' }] },
      navbar: { chosenCategory: 'popular', chosenGenre: 'genre' },
    })
    store.dispatch = jest.fn()

    component = create(
      <MemoryRouter>
        <Provider store={store}>
          <Navbar />
        </Provider>
      </MemoryRouter>,
    )
  })

  it('Should render and match snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('Should call dispatch after click', () => {
    act(() => {
      component.root.findAllByProps({ testid: 'test' })[0].props.onClick()
    })
    expect(store.dispatch).toHaveBeenCalledTimes(2)
  })
})
