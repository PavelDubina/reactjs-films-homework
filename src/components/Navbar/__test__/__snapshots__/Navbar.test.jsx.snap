// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Should render and match snapshot 1`] = `
<div
  className="navbar"
>
  <div
    className="navbar__buttons"
  >
    <div
      className="navbar__buttons_button active_button"
      onClick={[Function]}
      testid="test"
    >
      Trending
    </div>
    <div
      className="navbar__buttons_button"
      onClick={[Function]}
      testid="test"
    >
      Top Rated
    </div>
    <div
      className="navbar__buttons_button"
      onClick={[Function]}
      testid="test"
    >
      Upcoming
    </div>
  </div>
  <div
    className="select"
  >
    <div
      className="select__button"
      onClick={[Function]}
      testid="1"
    >
      <span
        className="select__button_text"
      />
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-chevron-circle-down fa-w-16 select__button_arrow"
        data-icon="chevron-circle-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256c0 137-111 248-248 248S8 393 8 256 119 8 256 8s248 111 248 248zM273 369.9l135.5-135.5c9.4-9.4 9.4-24.6 0-33.9l-17-17c-9.4-9.4-24.6-9.4-33.9 0L256 285.1 154.4 183.5c-9.4-9.4-24.6-9.4-33.9 0l-17 17c-9.4 9.4-9.4 24.6 0 33.9L239 369.9c9.4 9.4 24.6 9.4 34 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;
